cmd/merge-nodes/internal/merge_action.go

```diff
@@ -0,0 +1,230 @@
+package internal
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/signing"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/sql/localsql"
+)
+
+const (
+	localDbFile = "local.sql"
+
+	keyDir                  = "identities"
+	supervisedIDKeyFileName = "local.key"
+)
+
+func MergeDBs(ctx context.Context, dbLog *zap.Logger, from, to string) error {
+	// Open the target database
+	var dstDB *localsql.Database
+	var err error
+	dstDB, err = openDB(dbLog, to)
+	var schemaErr ErrInvalidSchemaVersion
+	switch {
+	case errors.As(err, &schemaErr):
+		dbLog.Error("target database has an invalid schema version - aborting merge",
+			zap.Int("db version", schemaErr.Actual),
+			zap.Int("expected version", schemaErr.Expected),
+		)
+		return err
+	case errors.Is(err, fs.ErrNotExist):
+		// target database does not exist, create it
+		dbLog.Info("target database does not exist, creating it", zap.String("path", to))
+		if err := os.MkdirAll(to, 0o700); err != nil {
+			return fmt.Errorf("create target directory: %w", err)
+		}
+		if err := os.MkdirAll(filepath.Join(to, keyDir), 0o700); err != nil {
+			return fmt.Errorf("create target key directory: %w", err)
+		}
+
+		dstDB, err = localsql.Open("file:"+filepath.Join(to, localDbFile),
+			sql.WithLogger(dbLog),
+		)
+		if err != nil {
+			return err
+		}
+		// -------------
+		// this code should not be needed but without it, Test_MergeDBs_Successful_Empty_Dir fails on INSERT initial_post
+		// with sqlite.SQLITE_CONSTRAINT_PRIMARYKEY error
+		if err := dstDB.Close(); err != nil {
+			return fmt.Errorf("close target database: %w", err)
+		}
+		dstDB, err = localsql.Open("file:"+filepath.Join(to, localDbFile),
+			sql.WithLogger(dbLog),
+		)
+		if err != nil {
+			return err
+		}
+		// -------------
```

Comment:
@dshulyak this just closes the DB and opens it again, but if I don't do this `Test_MergeDBs_Successful_Empty_Dir` fails on my machine with `sql.ErrObjectExists` caused by a `sqlite.SQLITE_CONSTRAINT_PRIMARYKEY`